src/PrestaShopBundle/Controller/Admin/Sell/Order/OrderController.php

```diff
@@ -246,13 +255,62 @@ public function viewAction(int $orderId): Response
         ], [
             'id_order' => $orderId,
         ]);
+        $addOrderProductForm = $this->createForm(AddOrderProductType::class);
+        $updateOrderProductForm = $this->createForm(UpdateOrderProductType::class);
 
         return $this->render('@PrestaShop/Admin/Sell/Order/Order/view.html.twig', [
             'showContentHeader' => false,
             'orderForViewing' => $orderForViewing,
             'addOrderCartRuleForm' => $addOrderCartRuleForm->createView(),
             'updateOrderStatusForm' => $updateOrderStatusForm->createView(),
             'addOrderPaymentForm' => $addOrderPaymentForm->createView(),
+            'addOrderProductForm' => $addOrderProductForm->createView(),
+            'updateOrderProductForm' => $updateOrderProductForm->createView(),
+        ]);
+    }
+
+    /**
+     * @param int $orderId
+     * @param int $orderCartRuleId
+     *
+     * @return RedirectResponse
+     */
+    public function removeCartRuleAction(int $orderId, int $orderCartRuleId): RedirectResponse
```

Comment:
Dont forget security

Reply:
It will be done in last PR before enabling page, many PRs modify this file, so I want to have as little as possible conflicts.